pkg/controllers/rollout/controller.go

```diff
@@ -709,6 +712,10 @@ func (r *Reconciler) SetupWithManager(mgr runtime.Manager) error {
 				klog.V(2).InfoS("Handling a clusterResourceOverrideSnapshot create event", "clusterResourceOverrideSnapshot", klog.KObj(e.Object))
 				handleClusterResourceOverrideSnapshot(e.Object, q)
 			},
+			DeleteFunc: func(ctx context.Context, e event.DeleteEvent, q workqueue.RateLimitingInterface) {
```

Comment:
trigger it when cx deletes the resourceOverride or clusterResourceOverride?

The current fix relies on the assumption and current implementation, when we delete the latest snapshot, it means cx is deleting override itself. If we prefer the current one, better to add a comment to record this.

another tiny diff is, if controller fails to delete resourceOverride for some reason, the override revert changes will be applied when the latestSnapshot is deleted.

Reply:
Although the diff is small, I thought about it a bit more and I think it's better to watch CRO/RO directly. I was a bit concerned that the rollout controller was watching too many types of resources but I think it should be okey.